cli/azd/pkg/infra/provisioning/bicep/bicep_provider_test.go

```diff
@@ -63,6 +63,24 @@ func TestBicepPlan(t *testing.T) {
 	)
 }
 
+func TestBicepPlanKeyVaultRef(t *testing.T) {
+	mockContext := mocks.NewMockContext(context.Background())
+	prepareBicepMocks(mockContext)
+	infraProvider := createBicepProvider(t, mockContext)
+
+	deploymentPlan, err := infraProvider.plan(*mockContext.Context)
+
+	require.Nil(t, err)
+
+	require.IsType(t, &deploymentDetails{}, deploymentPlan)
+	configuredParameters := deploymentPlan.CompiledBicep.Parameters
+
+	require.NotEmpty(t, configuredParameters["kvSecret"])
```

Comment:
Just noting this for anyone else who finds this slightly confusing: it took me awhile to figure out that these tests, as part of `createBicepProvider`, references the `test/functional/testdata/samples/webapp` samples, which is where `kvSecret` is getting the keyvault reference set.

In these tests, we also mock out bicep build implementation, hence why the `kvSecret` needs to be updated here. I could see why we do this to avoid invoking real `bicep build`, but I'm also not entirely sure if the mocks actually replicate real-world behavior in Bicep, but at least the tests seem to pass for now.